smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -306,11 +308,13 @@ private boolean writeRequestQueryString(
             Model model = context.getModel();
             for (HttpBinding binding : queryBindings) {
                 String memberName = symbolProvider.toMemberName(binding.getMember());
+                writer.addImport("extendedEncodeURIComponent", "__extendedEncodeURIComponent",
+                        "@aws-sdk/smithy-client");
                 writer.openBlock("if (input.$L !== undefined) {", "}", memberName, () -> {
                     Shape target = model.expectShape(binding.getMember().getTarget());
-                    String queryValue = getInputValue(context, binding.getLocation(), "input." + memberName,
-                            binding.getMember(), target);
-                    writer.write("query['$L'] = $L;", binding.getLocationName(), queryValue);
+                    String encodedQueryValue = "__extendedEncodeURIComponent(" + getInputValue(context,
+                            binding.getLocation(), "input." + memberName, binding.getMember(), target) + ")";
+                    writer.write("query['$L'] = $L;", binding.getLocationName(), encodedQueryValue);
```

Comment:
[Query locations also need to be percent-encoded](https://awslabs.github.io/smithy/spec/http.html#httpquery-trait), you can clean this up a little this way:

```java
String queryValue = getInputValue(context, binding.getLocation(), "input." + memberName,
        binding.getMember(), target);
writer.write("query[__extendedEncodeURIComponent($S)] = __extendedEncodeURIComponent($L);",
        binding.getLocationName(), encodedQueryValue);
```

Reply:
fixed in 2725e7d